pkg/kubelet/volume/plugins.go

```diff
@@ -80,6 +80,17 @@ type Host interface {
 
 	// GetKubeClient returns a client interface
 	GetKubeClient() client.Interface
+
+	// NewWrapperBuilder finds an appropriate plugin with which to handle
+	// the provided spec.  This is used to implement volume plugins which
+	// "wrap" other plugins.  For example, the "secret" volume is
+	// implemented in terms of the "emptyDir" volume.
+	NewWrapperBuilder(spec *api.Volume, podUID types.UID) (Builder, error)
```

Comment:
Sure, I think we can add that on after this.  Change these to take a VolumeSpecReader interface and implement a .GetSpec(key string) function.  For volume plugins, instead of:

```
if spec.GCEPersistentDisk != nil {
```

they would say:

```
if specReader.GetSpec("GCEPersistentDisk") != nil {
```

or something like that.  Another alternative would be to change `CanSupport(spec *api.VolumeSource)` into `CanSupport(spec interface{})` and make the probe loop do something like:

```
var i interface{} = FindTheFirstNonNil(spec)
for each plugin {
    if plugin.CanSupport(i)
}
```

Plugins could then try to convert i into their own Spec type.